vignettes/testing_demo.Rmd

```diff
@@ -1,51 +1,467 @@
+    divided by the 7-day average of new daily tests. For a handful of countries. 
+    OWID is able to get the test positivity rate directly from the same source as 
+    number of tests, and we use this pre-computed value for the positivity rate in 
+    these cases as it may be more reliable. Note they changed their 
+    algorithm in Jan 2022 to take any number of days between observations for interpolation, see 
+    [here](https://github.com/owid/covid-19-data/blob/master/scripts/scripts/testing/smoother.R). 
+    
+    + For FIND we manually interpolate the cumulative number of tests, new daily tests
+    and the 7-day average of new daily tests to match the OWID outputs.
+    FIND scrapes websites and testing data sources once daily for a given date, 
+    thus if the original source does not update, the scraped value is the same. 
+    To be precise, if a the scraped cumulative test number has not changed on 
+    subsequent days, we interpret this as no update to the number of tests (that 
+    is, the original value from FIND provided for "new tests" is 0). To interpolate
+    in a manner consistent with OWID, we first only consider an update to have
+    occurred if the cumulative testing number has actually changed. So if the 
+    values are recorded as 90, 100, 100, 100, 100, 200 for Sunday through Friday,
+    respectively, the interpolated series would be 90, 100, 125, 150, 175, 200. 
+    The maximum number of days we allow between two 
+    cumulative testing values is 31 days(note that this can be changed by the param 
+    `find_maxgap` in `get_testing_long). Test positivity is based on the ratio 
+    of 7-day average of new daily cases (from JHU) to 7-day average of new daily 
+    tests. 
+    
+- For each data source, we extract the **most recent available data point in the 
+past 14 days** for the 7-day average of new daily tests per 1000 and the test 
+positivity rate. 
+
+- We flag the data point if in the 7 days used for the calculation there was a 
+drop in the total number of tests, there was a sudden increase in the number of 
+tests suggesting a reporting anomaly (currently set to 5% daily increase in the 
+interpolated number of total tests), or there was a decrease in the total number 
+of cases (with exceptions for a small number countries in OWID where the test 
+positivity rate is not based on this number).
+
+- We examine countries with recent data points (in the last 14 days) and with no
+quality flags. 
+
+- Among these countries, if a country/area has data points in both FIND and OWID,
+we select OWID.
+
+- If a country has data, but none that pass the quality flags, the preferred 
+data source is set to NA.
+
+We report the recent values for each country/area along with the preferred data 
+source. Note that, it's plausible that one week to the next, the preferred data 
+source for a country could shift between FIND and OWID. 
+
+### More details on flags
```

Comment:
everything below this is verbatim per Bhezad (so could probably also use another read through:)